sql-cli/sql_cli/cli.py

```diff
@@ -34,6 +34,25 @@ def generate_dag(project: Project, env: str, workflow_name: str) -> Path:
         rprint(f"[bold red]The workflow {workflow_name} contains a cycle! {dag_cycle}[/bold red]")
         raise Exit(code=1)
     rprint("The DAG file", dag_file.resolve(), "has been successfully generated. 🎉")
+    import_errors = DagBag(process_subdir(str(dag_file))).import_errors
+    if import_errors:
+        all_errors = "\n\n".join(list(import_errors.values()))
```

Comment:
This is an additional great validation step. I am wondering if this alone could replace some of the exception handling like checking for dag cycles for example. But wondering how cryptic the output looks for (sql analyst) users.

Could you move that into a function, please? It looks very complex.

Reply:
@feluelle I think for some common use-cases it might be better for us to own the error handling so we can send analyst friendly error messages (e.g. an analyst shouldn't need an airflow stacktrace). that said I'll move the import error checking into function.